site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: blog-post
+title: Announcing FusionAuth 1.30.0
+description: The FusionAuth 1.30.0 releases a robust suite of Advanced Threat Detection features, a JWT vending machine, myriad webhook events, and more.
+author: Akira Brand
+image: blogs/release-1-30/product-update-fusionauth-1.30.png
+category: blog
+tags: topic-troubleshooting
+excerpt_separator: "<!--more-->"
+---
+
+We're excited to announce the release of version 1.30.0 of FusionAuth. This version shipped on August 12, 2021. 1.30.0 includes a robust Advanced Threat Detection feature, adds 38 webhook events, resolves issues for FusionAuth community members and customers, and more! 
+
+<!--more-->
+
+This release contained a number of features, enhancements, and bug fixes. Please see the [release notes](/docs/v1/tech/release-notes/#version-1-30-0) for a full breakdown of the changes between 1.29 and 1.30.0. 
+
+There are a few items worth calling out.
+
+## JWT Vending Machine
+
+There are scenarios where a user does not yet exist, or is in the process of being created and needs access to services. This may be an anonymous user, or a future user. In these cases we cannot yet authenticate to FusionAuth because the user does not yet exist, or they may not yet have a password.
+
+To solve this, we have created the possiblity of opening an API that allows a JWT to be created with a payload defined by the API called.  This would require an API key, so the caller must be privledged.  
+
+Here is an example of the endpoint at work: 
+
+Endpoint: /api/jwt
+
+```
+
+{
+  "claims": {
+     "roles": [ "awesome" ],
+     "id": 42
+   }
+}
+```
+
+Using this ability, a new user on your app can start interacting with the app as a unique user, before needing to create an account. Then, once they do create an account, their unique Id is translated onto their profile. 
```

Comment:
With this feature, an anonymous app visitor can interact with your platform and subsequently create an account and have all those interactions recorded (if desired).  

A common use case might be a visitor and a shopping cart.  You would not want to lose precious user browsing data while the user was evaluating your site.  Once they commit to a purchase and register, likely you will want to preserve portions of their initial visit to generate future revenue.